plugins/mysql/mysql_disk_by_user

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+# -*- python -*-
+# -*- coding: utf-8 -*-
+#
+# Plugin to monitor MySQL database disk usage per user
+#
+# Usage: Place in /etc/munin/plugins/ (or link it there using ln -s)
+#   Add this to your /etc/munin/plugin-conf.d/munin-node:
+#       [mysql_disk_size]
+#       user mysql
+#       users_with_underscores foo_bar d_ritchie # usernames of users in the
+#                                                # system that include an underscore
+#       limit 1024000 # minimum db size in order to report a specific user,
+#                     # in bytes (defaults to 5M). "no" for none.
+#
+# 2019-07-25: v 1.0 pcy <pcy.ulyssis.org>: created
+#
+# Parameters understood:
+#
+#   config   (required)
+#   autoconf (optional - used by munin-config)
+#
+# magic markers:
+#%# family=auto
+#%# capabilities=autoconf
+
+import os
+import re
+import subprocess as sp
+import sys
+
+
+def weakbool(x):
+    u = x.upper().strip()
+    if u == "FALSE" or u == "NO" or u == "N":
+        return False
+    try:
+        return bool(int(x))
+    except ValueError:
+        return bool(x)
+
+
+LIMIT = os.getenv('limit', str(50000 * 1024))
+try:
+    LIMIT = int(LIMIT) if weakbool(LIMIT) else None
+except ValueError:
+    LIMIT = None
+
+exceptions = os.getenv('users_with_underscores', '').split(' ')
+if exceptions == ['']:
+    exceptions = []
+
+
+def name_from_path(path):
+    filename = os.path.basename(path)
+    for name in exceptions:
+        if filename.startswith(name):
+            return name
+    name = filename.split('_')[0]
+
+    def decode_byte(m):
+        return bytes.fromhex(m.group(1)).decode('utf-8')
+
+    return re.sub('@00([0-9a-z]{2})', decode_byte, name)
+
+
+def sizes_by_name(limit=None):
+    sizes = dict({})
+    output = sp.run(['du', '-B1'], cwd='/var/lib/mysql', stdout=sp.PIPE)
```

Comment:
The path `/var/lib/mysql` should be configurable, I guess.

Additionally: wouldn't it be easier to access the files with python code and check their size without spawning `du`?

Reply:
Made it configurable.

Rewrote the `du` part in Python, but it's a bit longer now, actually.